include/ginkgo/core/base/mtx_io.hpp

```diff
@@ -118,7 +118,7 @@ enum class layout_type {
  */
 template <typename ValueType, typename IndexType>
 void write_raw(std::ostream& os, const matrix_data<ValueType, IndexType>& data,
-               layout_type layout = layout_type::array);
+               layout_type layout = layout_type::coordinate);
```

Comment:
I think we use array as default in the beginning to keep the same behavior as the old interface:
`void write_raw(std::ostream& os, const matrix_data<ValueType, IndexType>& data);`
but it should not be public interface break with this change